lib/instrument/transformer/nskeyed.js

```diff
@@ -0,0 +1,310 @@
+  DTTapHeartbeatMessage: () => { decodeArchive = DTTapMessagePlist; },
+  DTTapMessageArchive: () => { decodeArchive = DTTapMessagePlist; },
+  DTKTraceTapMessage: () => { decodeArchive = DTTapMessagePlist; },
+  ErrorArchive: () => { decodeArchive = NSError; },
+  ExceptionArchive: () => { decodeArchive = NSException; },
+  NSDictionary: () => { decodeArchive = DictArchive; },
+  NSMutableDictionary: () => { decodeArchive = DictArchive; },
+  NSArray: () => { decodeArchive = ListArchive; },
+  NSDate: () => {
+    decodeArchive = (archive) => uuidStringify(archive.decode('NS.uuidbytes'));
+    encodeArchive = (obj, archive) => archive._archiveObj['NS.uuidbytes'] = uuidParse(obj).buffer;
+  },
+  NSError: () => { decodeArchive = NSError; },
+  NSException: () => { decodeArchive = NSException; },
+  NSMutableArray: () => { decodeArchive = ListArchive; },
+  NSMutableSet: () => { decodeArchive = ListArchive; },
+  NSMutableString: () => { decodeArchive = (archive) => archive.decode('NS.string'); },
+  NSMutableData: () => { decodeArchive = (archive) => archive.decode('NS.data'); },
+  NSNull: () => { decodeArchive = () => null;},
+  NSSet: () => { decodeArchive = ListArchive; },
+  NSUUID: () => {
+    decodeArchive = (archive) => UNIX2APPLE_TIMESTAMP_SECOND + archive.decode('NS.time');
+    encodeArchive = (obj, archive) => archive.encode('NS.time', obj - UNIX2APPLE_TIMESTAMP_SECOND);
+  },
+  NSURL: () => {
+    decodeArchive = (archive) => ({$class: 'NSURL', base: archive.decode('NS.base'), relative: archive.decode('NS.relative')});
+    encodeArchive = (obj, archive) => {
+      archive.encode('NS.base', obj._base);
+      archive.encode('NS.relative', obj._relative);
+    };
+  },
+  XCTCapabilities: () => { decodeArchive = (archive) => archive.decode('capabilities-dictionary'); },
+});
+
+/**
+ * Capable of unpacking an archived object tree in the NSKeyedArchive format.
+ * Apple's implementation can be found here:
+ * https://github.com/apple/swift-corelibs-foundation/blob/main/Sources/Foundation/NSKeyedUnarchiver.swift
+ */
+class Unarchive {
+  constructor (inputBytes) {
+    this.input = inputBytes;
+    this.unpackedUids = {};
+    this.topUID = NULL_UID;
+    this.objects = [];
+  }
+
+  unpackArchiveHeader () {
+    const plist = plistlib.parseBuffer(this.input)[0];
+    if (plist.$archiver !== NSKEYEDARCHIVER) {
+      throw new Error(`unsupported encoder: ${plist.$archiver}`);
+    }
+    if (plist.$version !== NSKeyedArchiveVersion) {
+      throw new Error(`expected ${NSKeyedArchiveVersion}, got ${plist.$version }`);
+    }
+    const top = plist.$top;
+    const topUID = top.root;
+    if (!topUID) {
+      throw new Error(`top object did not have a UID! dump: ${JSON.stringify(top)}`);
+    }
+    this.topUID = topUID;
+    this.objects = plist.$objects;
+  }
+
+  /**
+   * use the UNARCHIVE_CLASS_MAP to find the unarchiving delegate of a uid
+   */
+  classForUid (index) {
```

Comment:
this method could probably return an instance instead of a class, since we anyway call `new` on its result afterwards